validator/sawtooth_validator/execution/processor_iterator.py

```diff
@@ -259,3 +259,40 @@ def remove_processor(self, processor_identity):
     def __len__(self):
         with self._lock:
             return len(self._processors)
+
+
+class _CurrentMaxMap(object):
+    """A dictionary that maps a key to a maximum value and a counter value"""
+    def __init__(self):
+        self._dict = dict()
+        self._lock = RLock()
+
+    def get(self, key, default=None):
+        with self._lock:
+            return self._dict.get(key)
+
+    def add_if_unique(self, key, maximum):
+        with self._lock:
+            if not self._dict.get(key):
```

Comment:
I would think that `if key not in self._dict` would be more meaningful and possibly more performant.